api/src/main/java/io/opentelemetry/trace/propagation/JaegerPropagator.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.propagation;
+
+import static io.opentelemetry.internal.Utils.checkNotNull;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.internal.StringUtils;
+import io.opentelemetry.trace.DefaultSpan;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TraceState;
+import io.opentelemetry.trace.TracingContextUtils;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Implementation of the Jaeger propagation protocol. See <a
+ * href=https://www.jaegertracing.io/docs/client-libraries/#propagation-format>Jaeger Propogation
+ * Format</a>.
+ */
+@Immutable
+public class JaegerPropagator implements HttpTextFormat {
+
+  private static final Logger logger = Logger.getLogger(HttpTraceContext.class.getName());
+
+  static final String TRACE_ID_HEADER = "uber-trace-id";
+  static final String PARENT_SPAN = "0";
+  static final String TRUE_INT = "1";
+  static final String FALSE_INT = "0";
```

Comment:
TRUE_INT and FALSE_INT kinda misleading names because type is String and they are using to mark that span is sampled, if u are wanna use static variable consider to use smth like IS_SAMPLED